models/articles.models.js

```diff
@@ -1,24 +1,32 @@
 const db = require("../db/connection.js");
 
-exports.selectArticles = () => {
-    return db.query(`
-        SELECT 
-            articles.article_id, 
-            articles.title, 
-            articles.topic,
-            articles.author,
-            articles.created_at, 
-            articles.votes, 
-            articles.article_img_url, 
-            COUNT(comments.comment_id) AS comment_count
-        FROM articles
-        LEFT JOIN comments ON articles.article_id = comments.article_id
-        GROUP BY articles.article_id
-        ORDER BY articles.created_at DESC;`)
+exports.selectArticles = (topic) => {
+    let queryString = `
+            SELECT 
+                articles.article_id, 
+                articles.title, 
+                articles.topic,
+                articles.author,
+                articles.created_at, 
+                articles.votes, 
+                articles.article_img_url, 
+                COUNT(comments.comment_id) AS comment_count
+            FROM articles
+            LEFT JOIN comments ON articles.article_id = comments.article_id `
+    
+    if (topic) {
+        queryString += `WHERE topic = '${topic}' `
```

Comment:
Don't think this will prevent SQL injection.  Perhaps a green list of accepted topics could help?